conduit/plugins/processors/filterprocessor/gen/generate.go

```diff
@@ -1,31 +1,222 @@
 package main
 
 import (
-	"bytes"
 	"fmt"
-	"go/format"
+	"io"
 	"os"
 	"reflect"
-	"sort"
 	"strings"
+	"text/template"
 
+	"github.com/algorand/indexer/conduit/plugins/processors/filterprocessor/gen/internal"
+
+	"github.com/algorand/go-algorand-sdk/v2/types"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
 	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/protocol"
 )
 
+// ignoreTags are things that we specifically want to exclude from the output.
+var ignoreTags = map[string]bool{
+	// this is a constant
+	"txn.gh": true,
+	// no point to filtering on a lease
+	"txn.lx": true,
+	// no point to filter on signatures
+	"sig":              true,
+	"msig.subsig":      true,
+	"lsig.sig":         true,
+	"lsig.arg":         true,
+	"lsig.l":           true,
+	"lsig.msig.subsig": true,
+	// no point in filtering on keys
+	"txn.votekey": true,
+	"txn.selkey":  true,
+	"txn.sprfkey": true,
+	// no point in filtering on state proof things?
+	"txn.sp.c":       true,
+	"txn.sp.S.pth":   true,
+	"txn.sp.S.hsh":   true,
+	"txn.sp.S.hsh.t": true,
+	"txn.sp.P.pth":   true,
+	"txn.sp.P.hsh":   true,
+	"txn.sp.P.hsh.t": true,
+	"txn.sp.r":       true,
+	"txn.sp.pr":      true,
+	"txn.spmsg.b":    true,
+	"txn.spmsg.v":    true,
+	// inner transactions are handled differently
+	"dt.itx": true,
+	// TODO: this can be removed if the sub fields are supported
+	"dt": true,
+	// TODO: support map types?
+	"dt.gd": true,
+	"dt.ld": true,
+	// TODO: support array types?
+	"txn.apaa": true,
+	"txn.apat": true,
+	"txn.apfa": true,
+	"txn.apbx": true,
+	"txn.apas": true,
+	"txn.apap": true,
+	"txn.apsu": true,
+	"dt.lg":    true,
+}
+
+func noCast(t reflect.StructField) bool {
+	switch reflect.New(t.Type).Elem().Interface().(type) {
+	case uint64:
+		return true
+	case int64:
+		return true
+	case string:
+		return true
+	}
+	return false
+}
+
+func simpleCast(t reflect.StructField) string {
+	switch reflect.New(t.Type).Elem().Interface().(type) {
+	// unsigned
+	case uint:
+		return "uint64"
+	case uint8:
+		return "uint64"
+	case uint16:
+		return "uint64"
+	case uint32:
+		return "uint64"
+	// signed
+	case int:
+		return "int64"
+	case int8:
+		return "int64"
+	case int16:
+		return "int64"
+	case int32:
+		return "int64" //
+	// alias
+	// SDK types
+	case types.MicroAlgos:
+		// SDK microalgo does not need ".Raw"
+		return "uint64"
+	case types.AssetIndex:
+		return "uint64"
+	case types.AppIndex:
+		return "uint64"
+	case types.Round:
+		return "uint64"
+	case types.OnCompletion:
+		return "uint64"
+	case types.StateProofType:
+		return "uint64"
+	case types.TxType:
+		return "string"
+	// go-algorand types
+	case basics.AssetIndex:
```

Comment:
do we need these types because we aren't using the sdk types in the importer?

Reply:
I'll remove the go-algorand types in a followup. I didn't want to change the behavior while refactoring.